### ISSUE TITLE
use mobile device timezone when sending "You're going oncall" push notifications

### ISSUE THREAD

**Issue by joeyorlando** [grafana/oncall#2450]
### What would you like to see!

Reported by @dannykopping [here](https://raintank-corp.slack.com/archives/C0229FD3CE9/p1688626673295129).

We use the OnCall user's stored timezone to display the oncall shift datetimes in the "You're going OnCall" push notifications. This leads to some issues where the user may want to have their Grafana data displayed in a certain timezone (eg. UTC), but have datetimes in their push notifications localized to their current location.

The easiest approach here is to have the mobile device tell the backend what this timezone is, and use this value instead when sending push notifications. 

**More details**
Currently this is how the OnCall backend receives the Grafana user's timezone. On web plugin load, we take the following value 👇 (from this [code here](https://github.com/grafana/oncall/blob/dev/grafana-plugin/src/models/user/user.ts#L60-L81)) and send it to the backend for persistance. This value is what's used to localize the shift datetimes in the push notification.
```tsx
const timezone = grafanaPreferencesTimezone === 'browser' ? dayjs.tz.guess() : grafanaPreferencesTimezone;
```

**Todos**
- [x] update the mobile app user settings object to add a `timezone` attribute
- [x] update "You're going OnCall" celery task logic to use the device's timezone instead of the one from the OnCall user

### Product Area

Mobile App

**Comment by Dieterbe** [comment]
doesn't this require https://github.com/grafana/oncall-mobile-app/pull/244 before we can close this?

**Comment by joeyorlando** [comment]
@Dieterbe maybe we close https://github.com/grafana/oncall-mobile-app/pull/244 as a duplicate and re-open this to track?

The intent of this issue this was to track the backend changes, which have been released. It should start working once https://github.com/grafana/oncall-mobile-app/pull/244 is released.

**Comment by Dieterbe** [comment]
> maybe we close grafana/oncall-mobile-app#244 as a duplicate 

that's a PR, not an issue.

> The intent of this issue this was to track the backend change

then i suggest making it clear in the title that this refers to backend only.
(though IMHO it seems maybe more useful to cover the oncall product as a whole in this ticket)

**Comment by joeyorlando** [comment]
ahh, I meant https://github.com/grafana/oncall/issues/2451, not https://github.com/grafana/oncall-mobile-app/pull/244.

closed #2451 as a duplicate of this issue. https://github.com/grafana/oncall-mobile-app/pull/244 has been marked to close this issue.

**Comment by imtoori** [comment]
the idea was to have #2451 for mobile and this one for backend 
no problem though
